### DISCUSSION TITLE
Opinions on coreboot

### DISCUSSION THREAD

**Discussion by carnage-mode** [fwupd/fwupd#7716] [General]
I was browsing through the device security page in the fwupd site, and I can't seem to find anything that comes with coreboot with a rating about HSI-1. Is it because coreboot is insecure or something?

(Note: Not a firmware guy)

Another thing I'd like to ask while we're here, Im having a hard time finding motherboards with HSI 2+, is there a way to sort by HSI ratings?

**Comment by hughsie** [comment]
Ohh, this is an answer I could expand out to a 1h talk at OSFC :) I'll try to be as brief and accurate as I can on such a complicated subject. I fear this answer might find it's way onto reddit and explode into a fireball, so what I'm saying here are my person views only.

I'll try to compare and contrast:

 * Coreboot is massively simpler in both scope and code than something like EDK (and I'll use this from this point onward to describe all the EDK-like IBV value-add offerings like AMIBIOS, InsydeH20 and PhoenixBios) but much harder for users to actually configure
* Coreboot, being simpler, has a much lower attack target and updates are fixed much quicker than in OEM BIOS images. I’d probably also say there’s more of a security response team for coreboot than EDK upstream these days. The IBV security teams are always trying to minimize vulnerability CVSS scores to avoid “bad publicity” and it’s almost impossible to work effectively *with* them.
* EDK provide a few features *some* enterprise users actually need, e.g. like network booting and things like Absolute Persistence -- which sometimes look a lot like a bad idea to the trained eye. Most of the value-add features are not used by most consumers IMHO.
 * Coreboot being an open source project has a much more (and better) code review than EDK, and in general, in my humble opinion is better quality code.
 * EDK is built by the IBV, ODM or OEM and is then signed by the platform key -- which makes it impossible to boot a 3rd party community BIOS (bad) -- but also makes it impossible to just flash a malicious BIOS from userspace. You _can_ use coreboot with BootGuard (to provide a platform root of trust) but I've never actually seem an image that actually does that.

So the tl;dr: is that coreboot fixes a lot of potential-security issues in EDK, but doesn't typically use BootGuard so any attacker can easily replace your open source bios with an equally-open-source, but, malicious image. From a “HSI score” point of view I don’t know what’s best – having a coreboot-image-without-BootGuard is better than a EDK-plus-IBV-value-add-with-BootGuard is better in some respects and worse in others.

Also, regards to sorting the HSI values; in some cases it's a bit fuzzy - i.e. the very same hardware can actually have different user-supplied HSI values depending on the specific SKU or workplace to user-configured options. So in both cases, it's complicated :)
